### ISSUE TITLE
build command

### ISSUE THREAD

**Issue by kenshaw** [gunk/gunk#93]
One of the goals of the project should be to have something like a `build` subcommand that analyzes the code in the passed package spec, and updates corresponding `.gunk` files with missing services. Think of this as a `.go -> .gunk` as opposed to `.gunk -> go`.

For example, given something like the following Go code:

```go
// +gunk:build brank.as/proto/v1/util
package util

import (
    goog "github.com/golang/protobuf/ptypes/empty"
)

// +gunk opt.Service
type Server struct {

}

func (s *Server) MyMethod(context.Context, *goog.Empty) (*goog.Empty, error) {
    return nil, nil
}
```

If the corresponding types / services in the `brank.as/proto/v1/util` package don't exist (ie, the `Server` type here, and the `MyMethod` method), then we should generate the corresponding definitions in `.gunk` files. Alternately, we can just build them and use it with the protoc toolchains.

I have not put much thought into this, but I do want 'reversibility' in some kind of fashion, to generate service definitions from the actual Go code.

**Comment by mvdan** [comment]
This reminds me of the current `convert` command, built by @vishen. It converts proto to Gunk, so perhaps we could also have it convert Go to Gunk - depending on what files it's given or it finds.

I'm not very fond of the `build` command name, because I'd naively expect that command to build the generated server/client code.

**Comment by kenshaw** [comment]
Yes, I agree `build` is a horrible name. It was the first that came to mind. Open to any suggestions. I don't think `convert` really conveys what its doing, however. It's more like `generate-inverse` or or even just `invert`.

**Comment by mvdan** [comment]
I think whatever name we settle with, the current `convert` command and this command should be the same. It would be confusing to have two commands that can translate X into Gunk files.

`convert` isn't great, but it's the best we've come up with so far. `invert` or `generate-inverse` sounds a bit wrong, because that seems like we could produce a Gunk file just from the generated Go/JS/whatever files. When in fact, the input is not the generated code, but the implementation of the server with special annotations.

**Comment by kenshaw** [comment]
Please note: I'm not sold on the `invert` name either. Just wanted to change the listing in the issue tracker.

**Comment by mvdan** [comment]
`extract` also seems fine to me. We don't have to stick with it forever, but it's certainly good enough for now.
